packages/@aws-cdk/custom-resources/README.md

```diff
@@ -14,10 +15,373 @@
 
 ---
 <!--END STABILITY BANNER-->
+- [Provider Framework](#provider-framework)
+  - [User-defined Handlers](#user-defined-handlers)
+    - [onEvent](#onevent)
+    - [isComplete](#iscomplete)
+  - [Provider API](#provider-api)
+  - [Execution Policy](#execution-policy)
+  - [Resources Instances](#resources-instances)
+  - [Patterns and Practices](#patterns-and-practices)
+    - [Wrapper constructs](#wrapper-constructs)
+    - [Packaging providers in nested stacks](#packaging-providers-in-nested-stacks)
+    - [Sharing providers across apps](#sharing-providers-across-apps)
+  - [Complete Example](#complete-example)
+- [Custom Resources for AWS APIs](#custom-resources-for-aws-apis)
+  - [Examples](#examples)
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
 
-#### AWS Custom Resource
+## Provider Framework
+
+The `@aws-cdk/custom-resources.Provider` is a "mini-framework" for
+implementing providers for AWS CloudFormation [custom
+resources](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html).
+It offers a high-level API which makes it easier to implement robust and
+powerful custom resources and includes the following capabilities:
+
+* Handles responses to AWS CloudFormation and protects against blocked
+  deployments
+* Validates values returned by the user handler to help with correct handler implementation
+* Supports asynchronous handlers to allow long operations which can exceed the AWS Lambda timeout
```

Comment:
How would you use the framework for an operation that takes around 20min to complete? This cannot be done in the `onEvent` handler (> 15min) so you would have to start this operation in your `onEvent` handler and immediately return with only a `PhysicalResourceId` (async invoke of another Lambda)?